### PR TITLE
Conflict editing: Say which commit failed to apply

### LINKED ISSUE #45: Feature request: Say which patch failed when editing a conflict
A useful feature of `git rebase -i` is that it says which commit failed to apply when there is a conflict:

    Could not apply badbeef... Commit bla bla

To me, this is essential information, because the editing is after all not of the end result, but some intermediate commit – you need to know which commit you're editing. Even if you can guess it, you don't want to take chances here, because of how easily this can make a tangled mess of the commits.

Git rebase also repeats the commit title inside the editable git conflict (whereas revise just puts "incoming" here). That would also suffice, though I really don't like the userfriendlyness of standard git conflicts. Thus why I suggest printing a message in plain English.

Perhaps the git conflict is the better place for this info, but then I would say the real problem is the readability of standard git conflicts: They look too much like a conflict between equals, and presents which two commits it came from, when what matters is which one commit you're editing. They really should just say `You are editing commit "Commit bla bla"`. If this is fixable, I would say you don't need the message, because then, a preview would hint even better to the user whether to edit or abort.

### PR DESCRIPTION
**Pull request by anordal**
Fixes #45

### CODE REVIEW

**Comment by anordal** [comment]
I changed the wording a little in the conflict file, but I wasn't so sure about it. What do you think?

* `incoming` → `this`: Short for "You are editing this commit".
* `new parent`/`old parent` → `new basis`/`old basis`: Parent obviously means parent commit in this case, but basis can stand on its own. It is the basis for the change.

**Comment by mystor** [comment]
> I changed the wording a little in the conflict file, but wasn't too sure about it. What do you think?
> 
>     * `incoming` → `this`: Short for "You are editing this commit".

"incoming" definitely wasn't the best name for it, but I'm not sure about "this" either. How about "current"?

>     * `new parent`/`old parent` → `new basis`/`old basis`: To say it without jargon, this is the "basis for the change".

I think I prefer the old wording. I find "basis" quite jargony as well, and "parent" is a more common word, also used by git. In my experience it hasn't been a source of confusion for people using git-revise.

File: gitrevise/merge.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def merge_blobs(
 
         # At this point, we know that there are merge conflicts to resolve.
         # Prompt to try and trigger manual resolution.
-        print(f"Merge conflict for '{path}'")
+        print(f"Merge conflict for '{path}', commit '{labels[2]}'")
```
Comment (mystor) [inline gitrevise/merge.py]: We can get away with multi-line prompts, so perhaps let's put the path & commit onto separate lines. Perhaps something like this?

```suggestion
        print(f"Merge conflict while rebasing '{labels[2]}'")
        print(f"  Path: '{path}'")
```

Comment (anordal) [reply gitrevise/merge.py]: Good idea, let's split it.

So you don't want to invent the term "rebase conflict"?
Or avoid calling it anything, like git rebase: `Could not apply '{labels[2]}'`?

Comment (anordal) [reply gitrevise/merge.py]: I used git revise, and changed the commit to this; what do you think?

```suggestion
-        print(f"Merge conflict for '{path}', commit '{labels[2]}'")
+        print(f"Conflict in applying '{labels[2]}'")
+        print(f"  Path: '{path}'")
```

Comment (mystor) [reply gitrevise/merge.py]: Works for me :-)
Thanks!
